addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -801,6 +801,73 @@ def test_message_route_alias_multiple_new(self):
         new_simple = self.env['mail.test.gateway'].search([('name', '=', 'Test Subject')])
         self.assertEqual(len(new_simple), 1, 'message_process: a new mail.test should have been created')
 
+    @mute_logger('odoo.addons.mail.models.mail_thread', 'odoo.models')
+    def test_mail_catchall_domain_allowed_validation(self):
```

Comment:
:socks: in the same file, there is a class 'TestMailAlias' holding alias-specific tests, this one could be moved in that class :)  like 'test_alias_domain_allowed_validation' to matche current naming in that class